### PR TITLE
Remove "using namespace" from header and move Instruction to dev::eth.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

File: libevmasm/AssemblyItem.h [hunk 1]
```diff
@@ -21,14 +21,13 @@
...
 
-#include <iostream>
-#include <sstream>
-#include <libdevcore/Common.h>
-#include <libdevcore/Assertions.h>
 #include <libevmasm/Instruction.h>
+#include <libevmasm/Exceptions.h>
 #include <liblangutil/SourceLocation.h>
-#include "Exceptions.h"
-using namespace dev::solidity;
```
Comment (chriseth) [inline libevmasm/AssemblyItem.h:31]: This is where it all started....

File: libevmasm/Instruction.h [hunk 2]
```diff
@@ -21,14 +21,14 @@
...
+#include <libevmasm/Exceptions.h>
 #include <libdevcore/Common.h>
 #include <libdevcore/Assertions.h>
-#include "Exceptions.h"
+#include <functional>
 
 namespace dev
 {
-namespace solidity
+namespace eth
```
Comment (chriseth) [inline libevmasm/Instruction.h:31]: And that's the other main change.

File: libyul/backends/evm/EVMAssembly.cpp [hunk 3]
```diff
@@ -43,31 +44,31 @@ void EVMAssembly::setSourceLocation(SourceLocation const&)
 	// Ignored for now;
 }
 
-void EVMAssembly::appendInstruction(solidity::Instruction _instr)
+void EVMAssembly::appendInstruction(dev::eth::Instruction _instr)
```
Comment (leonardoalt) [inline libyul/backends/evm/EVMAssembly.cpp:47]: `dev::eth` redundant here?

Comment (leonardoalt) [reply libyul/backends/evm/EVMAssembly.cpp:47]: This file is sometimes using `dev::eth` and sometimes not.

Comment (chriseth) [reply libyul/backends/evm/EVMAssembly.cpp:47]: Most of the places would clash with `yul::Instruction`.

File: libyul/backends/evm/NoOutputAssembly.cpp [hunk 4]
```diff
@@ -30,25 +30,25 @@ using namespace langutil;
 using namespace yul;
 
 
-void NoOutputAssembly::appendInstruction(solidity::Instruction _instr)
+void NoOutputAssembly::appendInstruction(dev::eth::Instruction _instr)
 {
-	m_stackHeight += solidity::instructionInfo(_instr).ret - solidity::instructionInfo(_instr).args;
+	m_stackHeight += instructionInfo(_instr).ret - instructionInfo(_instr).args;
```
Comment (leonardoalt) [inline libyul/backends/evm/NoOutputAssembly.cpp:35]: Why doesn't `instructionInfo` need the namespace?

File: libyul/backends/evm/EVMAssembly.cpp (hunk 3 above)
Comment (leonardoalt) [reply libyul/backends/evm/EVMAssembly.cpp:47]: ah ok

File: libyul/backends/evm/NoOutputAssembly.cpp (hunk 4 above)
Comment (chriseth) [reply libyul/backends/evm/NoOutputAssembly.cpp:35]: Ah. Weird.

Comment (chriseth) [reply libyul/backends/evm/NoOutputAssembly.cpp:35]: Ah, it's an instance of Koenig lookup.

Comment (chriseth) [reply libyul/backends/evm/NoOutputAssembly.cpp:35]: Do you want me to qualify the access?

**Comment by chriseth** [comment]
Fixed the tests (hopefully).

File: libyul/backends/evm/NoOutputAssembly.cpp (hunk 4 above)
Comment (leonardoalt) [reply libyul/backends/evm/NoOutputAssembly.cpp:35]: Not needed

**Review by leonardoalt — APPROVED** [review]

**Pull request merged**